xdsl/transforms/transform_interpreter.py

```diff
@@ -0,0 +1,43 @@
+"""Transform interpreter."""
+
+from dataclasses import dataclass
+
+from xdsl.dialects import builtin, transform
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.transform import TransformFunctions
+from xdsl.passes import Context, ModulePass
+from xdsl.utils.exceptions import PassFailedException
+
+
+@dataclass(frozen=True)
+class TransformInterpreterPass(ModulePass):
+    """Transform dialect interpreter"""
+
+    # TODO: Add the rest of options to the transform interpreter
+    # and add semantics for all of the other transform operations
+
+    name = "transform-interpreter"
+
+    entry_point: str = "__transform_main"
+
+    @staticmethod
+    def find_transform_entry_point(
+        root: builtin.ModuleOp, entry_point: str
+    ) -> transform.NamedSequenceOp:
+        for op in root.walk():
+            if (
+                isinstance(op, transform.NamedSequenceOp)
+                and op.sym_name.data == entry_point
+            ):
+                return op
+        raise PassFailedException(
+            f"{root} could not find a nested named sequence with name: {entry_point}"
+        )
```

Comment:
it feels like we should be able to use the SymbolTable mechanism here instead of walking the whole module, or something that more directly looks first for the right module, then looks up the function by name

Reply:
This mimics the implementation in MLIR. Can you specify a nested module where the nested module has no symbol? I.e.,

```
module {
  module {
    transform.named_sequence @__main() { // <--- this one?
    }
  }
}
```

Reply:
Oh interesting I'm not sure why they did it that way but I'm happy to mirror their mechanism.